vunit/json4vhdl.py

```diff
@@ -0,0 +1,62 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Copyright (c) 2014-2018, Lars Asplund lars.anders.asplund@gmail.com
+
+"""
+json4vhdl helper functions
+"""
+
+import json
+
+
+def encode_json(obj):
+    """
+    Convert object to stringified JSON
+
+    :param obj: Object to stringify
+
+    :example:
+
+    .. code-block:: python
+
+       stringified_generic = encode_json(generics)
+    """
+    return json.dumps(obj, separators=(',', ':'))
+
+
+def read_json(filename):
+    """
+    Read a JSON file and return an object
+
+    :param filename: The name of the file to read
+
+    :example:
+
+    .. code-block:: python
+
+       generics = read_json(join(root, "src/test/data/data.json"))
+    """
+    return json.loads(open(filename, 'r').read())
+
+
+def add_array_lens(obj):
+    """
```

Comment:
add_array_lens should not be added.

Reply:
Does this mean that it should be added to JSON-for-VHDL prior to merging this PR? Or will this go forward with a note explaining that the user must ensure that the first element in each array is the length of it?

Reply:
I think either fix it in json4vhdl first or do not have an example that needs dynamically sized arrays.
I do not think we should add work-arounds to the Public API of VUnit just because a feature is missing in another open source library. 

If you still want an example of this now even before the feature is added to json4vhdl it could be a bit more clear not add the array length as an element of the array but as a separate field in parallel with the array such as.

```json
{
"length": 3,
"data": [1,2,3]
}
```